engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/execution/CommandExecutionEngine.scala

```diff
@@ -59,8 +76,23 @@ class CommandExecutionEngine(interpreterContext: InterpreterContext)
     )
 
   /** @inheritdoc */
-  def invoke(cmd: Command): cmd.Result[Completion] = {
-    cmd.execute(runtimeContext, commandExecutionContext)
+  def invoke(cmd: Command): Future[Completion] = {
```

Comment:
Is it guaranteed the order of commands sent to `invoke(cmd)` method is the same as messages read from the web socket stream?

If not, then the `blockingQueue.add(c)` would still not guarantee the order...

Reply:
It is, yes.